### PR TITLE
Add AWSMachinePool test to eks e2e test

### LINKED ISSUE #2865: [e2e] Add `AWSMachinePool` to the EKS e2e test
/kind feature
/area provider/eks
/area testing
/priority important-soon
/help

**Describe the solution you'd like**
Currently the EKS e2e tests do not test the `AWSMachinePool`. We need to add a test to ensure there are no regressions in this area. 

**Anything else you would like to add:**
Suggestion is to add a new step in the tests which will add a AWSManagedMachinePool.

**Environment:**

- Cluster-api-provider-aws version: 1.0.0

### PR DESCRIPTION
**Pull request by pydctw**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
EKS e2e do not test AWSMachinePool flavor. This is an important flavor to support so added e2e test.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2865 

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits
- [x] includes documentation
- [ ] adds unit tests
- [x] adds or updates e2e tests

### CODE REVIEW

**Comment by richardcase** [comment]
Thanks for this @pydctw :tada: 

Just a small nit but:

/approve

I can lgtm to if you want to do a change for the nit.

**Comment by pydctw** [comment]
> Just a small nit but:
> 
> /approve
> 
> I can lgtm to if you want to do a change for the nit.

@richardcase, did you submit the comments? I don't see them. Once I get them, will make the changes.

File: test/e2e/suites/managed/helpers.go [hunk 1]
```diff
@@ -42,7 +43,8 @@ const (
 	EKSControlPlaneOnlyFlavor          = "eks-control-plane-only"
 	EKSControlPlaneOnlyWithAddonFlavor = "eks-control-plane-only-withaddon"
 	EKSMachineDeployOnlyFlavor         = "eks-machine-deployment-only"
-	EKSManagedPoolOnlyFlavor           = "eks-managed-machinepool-only"
+	EKSManagedMMachinePoolOnlyFlavor   = "eks-managed-machinepool-only"
```
[the remark this answers is not in the document]
Comment (richardcase) [reply test/e2e/suites/managed/helpers.go:46]: @pydctw - its just this comment.

[the remark this answers is not in the document]
Comment (pydctw) [reply test/e2e/suites/managed/helpers.go]: done

**Pull request merged**